src/lsm/lsm_cursor.c

```diff
@@ -887,29 +926,17 @@ __clsm_next(WT_CURSOR *cursor)
 
 	/* If we aren't positioned for a forward scan, get started. */
 	if (clsm->current == NULL || !F_ISSET(clsm, WT_CLSM_ITERATE_NEXT)) {
-		F_CLR(clsm, WT_CLSM_MULTIPLE);
 		WT_FORALL_CURSORS(clsm, c, i) {
 			if (!F_ISSET(cursor, WT_CURSTD_KEY_SET)) {
 				WT_ERR(c->reset(c));
 				ret = c->next(c);
-			} else if (c != clsm->current) {
-				c->set_key(c, &cursor->key);
-				if ((ret = c->search_near(c, &cmp)) == 0) {
-					if (cmp < 0)
-						ret = c->next(c);
-					else if (cmp == 0) {
-						if (clsm->current == NULL)
-							clsm->current = c;
-						else
-							F_SET(clsm,
-							    WT_CLSM_MULTIPLE);
-					}
-				} else
-					F_CLR(c, WT_CURSTD_KEY_SET);
-			}
+			} else if (c != clsm->current && (ret =
+			    __clsm_position_chunk(clsm, c, true, &cmp)) == 0 &&
+			    cmp == 0 && clsm->current == NULL)
+				clsm->current = c;
 			WT_ERR_NOTFOUND_OK(ret);
 		}
-		F_SET(clsm, WT_CLSM_ITERATE_NEXT);
+		F_SET(clsm, WT_CLSM_ITERATE_NEXT | WT_CLSM_MULTIPLE);
```

Comment:
The effect of this is that the first step of an iteration will check whether multiple cursors are pointing at the same value.  I think that is a reasonable simplification that won't cost us measurable performance.